src/main/scala/Day05Copy.scala

```diff
@@ -0,0 +1,40 @@
+import scala.annotation.tailrec
+import scala.io.*
+object Day05Copy extends App:
+  val day: String = this.getClass.getName.drop(3).init
+  val start1: Long = System.currentTimeMillis
+
+  private val polymer: String =
+    Source
+      .fromResource(s"input$day.txt")
+      .getLines
+      .next
+  def reduceString(str: String): String = {
+    @tailrec
+    def go(s: String, prefix : String = ""): String = {
+      @tailrec
+      def go2(i : Int) : Option[Int] = {
+        if(i >= s.length - 1) None
+        else if (s(i) == s(i + 1) + 32 || s(i) == s(i + 1) - 32) Some(i) // ascii difference between upper/lower
```

Comment:
could employ `toLowerCase` on `Char`s

Reply:
alternatively - consider loading the polymer as a `List[Char]` instead of `String`

this makes it possible to pattern match in the recursion, eg.:

```
s match
  case c1 :: c2 :: rest => ...
  case ...
```

in which case you don't need to index the string